--- a/crypto/ui/Makefile	2015-12-03 09:00:13.000000000 -0600
+++ b/crypto/ui/Makefile	2015-12-23 16:21:16.387256585 -0600
@@ -21,8 +21,8 @@
 COMPATOBJ= ui_compat.o
 
 LIB=$(TOP)/libcrypto.a
-LIBSRC= ui_err.c ui_lib.c ui_openssl.c ui_util.c $(COMPATSRC)
-LIBOBJ= ui_err.o ui_lib.o ui_openssl.o ui_util.o $(COMPATOBJ)
+LIBSRC= ui_err.c ui_lib.c ui_openssl.c ui_util.c ui_pase.c $(COMPATSRC)
+LIBOBJ= ui_err.o ui_lib.o ui_openssl.o ui_util.o ui_pase.o $(COMPATOBJ)
 
 SRC= $(LIBSRC)
 
