*** a/crypto/rand/Makefile	2015-12-03 09:00:09.000000000 -0600
--- b/crypto/rand/Makefile	2015-12-24 17:54:41.259417590 -0600
***************
*** 18,26 ****
  
  LIB=$(TOP)/libcrypto.a
  LIBSRC=md_rand.c randfile.c rand_lib.c rand_eng.c rand_err.c rand_egd.c \
! 	rand_win.c rand_unix.c rand_os2.c rand_nw.c
  LIBOBJ=md_rand.o randfile.o rand_lib.o rand_eng.o rand_err.o rand_egd.o \
! 	rand_win.o rand_unix.o rand_os2.o rand_nw.o
  
  SRC= $(LIBSRC)
  
--- 18,26 ----
  
  LIB=$(TOP)/libcrypto.a
  LIBSRC=md_rand.c randfile.c rand_lib.c rand_eng.c rand_err.c rand_egd.c \
! 	rand_win.c rand_unix.c rand_os2.c rand_nw.c rand_pase.c
  LIBOBJ=md_rand.o randfile.o rand_lib.o rand_eng.o rand_err.o rand_egd.o \
! 	rand_win.o rand_unix.o rand_os2.o rand_nw.o rand_pase.o
  
  SRC= $(LIBSRC)
  
