--- a/crypto/rand/rand_unix.c	2015-12-03 08:59:07.000000000 -0600
+++ b/crypto/rand/rand_unix.c	2015-12-22 16:56:55.711445819 -0600
@@ -116,7 +116,7 @@
 #include <openssl/rand.h>
 #include "rand_lcl.h"
 
-#if !(defined(OPENSSL_SYS_WINDOWS) || defined(OPENSSL_SYS_WIN32) || defined(OPENSSL_SYS_VMS) || defined(OPENSSL_SYS_OS2) || defined(OPENSSL_SYS_VXWORKS) || defined(OPENSSL_SYS_NETWARE))
+#if !(defined(OPENSSL_SYS_WINDOWS) || defined(OPENSSL_SYS_WIN32) || defined(OPENSSL_SYS_VMS) || defined(OPENSSL_SYS_OS2) || defined(OPENSSL_SYS_VXWORKS) || defined(OPENSSL_SYS_NETWARE) || defined(OPENSSL_SYS_PASE))
 
 # include <sys/types.h>
 # include <sys/time.h>
@@ -330,7 +330,8 @@
                                  * defined(OPENSSL_SYS_VMS) ||
                                  * defined(OPENSSL_SYS_OS2) ||
                                  * defined(OPENSSL_SYS_VXWORKS) ||
-                                 * defined(OPENSSL_SYS_NETWARE)) */
+                                 * defined(OPENSSL_SYS_NETWARE) ||
+                                 * defined(OPENSSL_SYS_PASE)) */
 
 #if defined(OPENSSL_SYS_VXWORKS)
 int RAND_poll(void)
